helix-term/src/commands/typed.rs

```diff
@@ -1225,6 +1225,35 @@ fn lsp_restart(
     Ok(())
 }
 
+fn lsp_stop(
+    cx: &mut compositor::Context,
+    _args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (_view, doc) = current!(cx.editor);
+
+    let ls_id = doc
+        .language_server()
+        .map(|ls| ls.id())
+        .context("LSP not running for the current document")?;
+
+    let config = doc
+        .language_config()
+        .context("LSP not defined for the current document")?;
+    cx.editor.language_servers.stop(&config);
+
+    cx.editor
+        .documents_mut()
+        .filter(|doc| doc.language_server().map_or(false, |ls| ls.id() == ls_id))
+        .for_each(|doc| doc.set_language_server(None));
```

Comment:
This is a small style nit but it's considered more idiomatic to use a `for` loop in cases like this:

```
for doc in cx.editor.documents_mut() {
    if doc.language_server().map_or(false, |ls| ls.id() == ls_id) {
        doc.set_language_server(None);
    }
}
```

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each

Reply:
Each of these documents should also have their `diagnostics` cleared

Reply:
https://github.com/helix-editor/helix/pull/5964/commits/19371699738cf66770c4af4d8a7fe2af708a5279